src/theory/sets/theory_sets_rels.cpp

```diff
@@ -32,24 +32,22 @@ typedef std::map< Node, std::unordered_set< Node, NodeHashFunction > >::iterator
 typedef std::map< Node, std::map< kind::Kind_t, std::vector< Node > > >::iterator               TERM_IT;
 typedef std::map< Node, std::map< Node, std::unordered_set< Node, NodeHashFunction > > >::iterator   TC_IT;
 
-TheorySetsRels::TheorySetsRels(context::Context* c,
-                               context::UserContext* u,
-                               eq::EqualityEngine* eq,
-                               TheorySetsPrivate& set)
-    : d_eqEngine(eq),
-      d_sets_theory(set),
-      d_trueNode(NodeManager::currentNM()->mkConst<bool>(true)),
-      d_falseNode(NodeManager::currentNM()->mkConst<bool>(false)),
-      d_shared_terms(u),
-      d_satContext(c)
+TheorySetsRels::TheorySetsRels(SolverState& s,
+                               InferenceManager& im,
+                               eq::EqualityEngine& e,
+                               context::Context* c,
+                               context::UserContext* u)
+    : d_state(s), d_im(im), d_ee(e), d_shared_terms(u)
 {
-  d_eqEngine->addFunctionKind(PRODUCT);
-  d_eqEngine->addFunctionKind(JOIN);
-  d_eqEngine->addFunctionKind(TRANSPOSE);
-  d_eqEngine->addFunctionKind(TCLOSURE);
-  d_eqEngine->addFunctionKind(JOIN_IMAGE);
-  d_eqEngine->addFunctionKind(IDEN);
-  d_eqEngine->addFunctionKind(APPLY_CONSTRUCTOR);
+  d_trueNode = NodeManager::currentNM()->mkConst(true);
+  d_falseNode = NodeManager::currentNM()->mkConst(false);
```

Comment:
`d_trueNode` and `d_falseNode` are redundant. They are already privately defined in `TheorySetsPrivate` as `d_true` and `d_false`. How about making them public constants in `TheorySetsPrivate` or better move them to  `Node` class which I think is is the ideal place for them?